servlet/src/main/java/com/linecorp/armeria/server/servlet/DefaultServletContext.java

```diff
@@ -0,0 +1,591 @@
+        requireNonNull(contextPath, "contextPath");
+        this.contextPath = contextPath;
+        this.servletContextName = contextPath.replace("/","");
+    }
+
+    /**
+     * Is server started.
+     */
+    public boolean isStarted() {
+        return started;
+    }
+
+    /**
+     * Set server started.
+     */
+    public void setStarted(boolean started) {
+        this.started = started;
+    }
+
+    /**
+     * Add a new mime mapping.
+     */
+    public void addMimeMapping(MimeMappings mimeMappings) {
+        requireNonNull(mimeMappings, "mimeMappings");
+        this.mimeMappings  = mimeMappings;
+    }
+
+    /**
+     * Get server address.
+     */
+    public InetSocketAddress getServerAddress() {
+        return serverAddress;
+    }
+
+    /**
+     * Set server address.
+     */
+    public void setServerAddress(InetSocketAddress serverAddress) {
+        requireNonNull(serverAddress, "serverAddress");
+        this.serverAddress = serverAddress;
+    }
+
+    /**
+     * Get servlet path.
+     */
+    public String getServletPath(String absoluteUri) {
+        requireNonNull(absoluteUri, "absoluteUri");
+        return servletUrlMapper.getServletPath(absoluteUri);
+    }
+
+    /**
+     * Get session timeout.
+     */
+    public int getSessionTimeout() {
+        return sessionTimeout;
+    }
+
+    /**
+     * Set session timeout.
+     */
+    public void setSessionTimeout(int sessionTimeout) {
+        checkArgument(sessionTimeout > 0, "sessionTimeout: %s (expected: > 0)", sessionTimeout);
+        this.sessionTimeout = sessionTimeout;
+    }
+
+    @Override
+    public String getContextPath() {
+        return contextPath;
+    }
+
+    @Override
+    public DefaultServletContext getContext(String uripath) {
+        requireNonNull(uripath, "uripath");
+        return this;
+    }
+
+    @Override
+    public int getMajorVersion() {
+        return 3;
+    }
+
+    @Override
+    public int getMinorVersion() {
+        return 0;
+    }
+
+    @Override
+    public int getEffectiveMajorVersion() {
+        return 3;
+    }
+
+    @Override
+    public int getEffectiveMinorVersion() {
+        return 0;
+    }
+
+    @Override
+    public String getMimeType(String file) {
+        requireNonNull(file, "file");
+        final int period = file.lastIndexOf('.');
+        if (period < 0) {
+            return null;
+        }
+        final String extension = file.substring(period + 1);
+        if (extension.length() < 1) {
+            return null;
+        }
+        return mimeMappings.get(extension);
+    }
+
+    @Override
+    public Set<String> getResourcePaths(String path) {
+        return null;
+    }
+
+    @Override
+    public URL getResource(String path) throws MalformedURLException {
+        return null;
+    }
+
+    @Override
+    public InputStream getResourceAsStream(String path) {
+        return null;
+    }
+
+    @Override
+    public String getRealPath(String path) {
+        return path;
```

Comment:
Yes, we can. I think we can do a simple resource management. We need a resource directory on linux such as: `/home/user/deploy/armeria_server/app/share` . When web browser send URL with .html | .css | .js | .jpg .... at last such as: `http://host:port/app/index.html`, getRealPath() will return file path `/home/user/deploy/armeria_server/app/share/index.html` then ServletHttpResponse will write FileOutputStream to web browser.

Reply:
Also, we have some issues: 
- How to read fast resource file, should cache resource file bytes on memory?
- How to combine multi html file such as: template file + menu file + body file
- Should we support Java server page standard (JSP) ? My opinion is shouldn't, because it's very bulky
I think we don't support resource management now. I will separate another PR to do this.

Reply:
> How to read fast resource file, should cache resource file bytes on memory?

Yes, we can cache the resources. Or we might use `HttpFile` but not sure it's going to work or not. Anyway, I think it's also deserved to have its own PR.

> How to combine multi html file such as: template file + menu file + body file?

I think we don't have to consider that right now.

> Should we support Java server page standard (JSP)? My opinion is shouldn't, because it's very bulky

I respect your opinion. 😄 

> I think we don't support resource management now. I will separate another PR to do this.

Yes, that's a good idea. Implementing servlet is really huge work. So let's break one by one. 😆